motor-control/firmware/motor_encoder_hardware.c

```diff
@@ -0,0 +1,48 @@
+//Initialize Header Files
+#include "motor_encoder_hardware.h"
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "platform_specific_hal_conf.h"
+
+void motor_encoder_set_pin(void* port, uint16_t pin, uint8_t active_setting)
+{
+    if(!port) {return;}
+    // Hall Function here
+    HAL_GPIO_WritePin(port, pin, active_setting);
+}
+
+static uint8_t invert_gpio_value(uint8_t setting) {
+    switch(setting) {
+        case GPIO_PIN_SET:
+            return GPIO_PIN_RESET;
+        case GPIO_PIN_RESET:
+            return GPIO_PIN_SET;
+        default:
+            return GPIO_PIN_SET;
+    }
+}
+
+void motor_encoder_reset_pin(void* port, uint16_t pin, uint8_t active_setting) {
```

Comment:
do we need this function, since these are all input pins?

Reply:
nope we don't need this.